src/testRunner/unittests/services/organizeImports.ts

```diff
@@ -168,6 +168,28 @@ namespace ts {
                 const expectedCoalescedImports = sortedImports;
                 assertListEqual(actualCoalescedImports, expectedCoalescedImports);
             });
+
+            it("Combine type-only imports separately from other imports", () => {
+                const sortedImports = parseImports(
+                    `import type { x } from "lib";`,
+                    `import type { y } from "lib";`,
+                    `import { z } from "lib";`);
+                const actualCoalescedImports = OrganizeImports.coalesceImports(sortedImports);
+                const expectedCoalescedImports = parseImports(
+                    `import { z } from "lib";`,
+                    `import type { x, y } from "lib";`);
+                assertListEqual(actualCoalescedImports, expectedCoalescedImports);
+            });
+
+            it("Do not combine type-only default, namespace, or named imports with each other", () => {
+                const sortedImports = parseImports(
+                    `import type { x } from "lib";`,
+                    `import type * as y from "lib";`,
+                    `import type z from "lib";`);

```

Comment:
Why cant this mix with `import type { x, default as z} from "lib` just like normal import statement ?

Reply:
I believe we decided it would be ambiguous what the "type" keyword applied to and forced explicitness.

Reply:
It could, but I’m not confident that’s desirable. My personal preference for readability would probably be what’s represented here in the test.

Reply:
@andrewbranch Consider adding a comment about this: `// Type-only imports are not allowed to combine (for readability)`.  Maybe on the test too.

Reply:
Probably possible at a later pass, but seems reasonable to leave it the way it is now until we get further feedback.

Reply:
If this is a stylistic choice in organizeImports, I don't think "allowed" is the right word - I definitely read that as a language restriction.

Reply:
My comment in the implementation was actually referring to the language restriction, ignoring the possibility of rewriting the default as a named import. I’ll follow up to clarify.